### DISCUSSION TITLE
PWA (Progressive Web App) Template

### DISCUSSION THREAD

**Discussion by hybridherbst** [needle-tools/needle-engine-support#17] [Demo and Starter Ideas]
PWAs are websites that can be installed and run offline.
They effectively look and feel like apps on their respective operating system (e.g. Windows, Android, Linux).

https://web.dev/progressive-web-apps/

A template to get started quickly with PWAs would have the following requirements:
- can be installed
- can run offline (all relevant files are cached)
- will update itself once online again and cache new files

**Comment by ROBYER1** [comment]
Got a little example of this here, caches fine on Windows and Android to work offline, if I figure out why it doesn't cache on IOS I'll update 
https://demo.epm.digital/pwatest/

Example project if anyone wants it:
[PWA-230116_uw.zip](https://github.com/needle-tools/needle-engine-support/files/10428280/PWA-230116_uw.zip)

**Reply by marwie** [reply]
Thank you for sharing! You used this vite plugin, right? https://github.com/vite-pwa/vite-plugin-pwa

**Reply by ROBYER1** [reply]
> Thank you for sharing! You used this vite plugin, right? https://github.com/vite-pwa/vite-plugin-pwa

Correct :) it wasn't actually 'zero config' as it said though and it doesn't work on iOS which is odd.
